x/clock/abci.go

```diff
@@ -0,0 +1,42 @@
+package clock
+
+import (
+	"log"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/CosmosContracts/juno/v17/x/clock/keeper"
+	"github.com/CosmosContracts/juno/v17/x/clock/types"
+)
+
+// EndBlocker executes on contracts at the end of the block.
+func EndBlocker(ctx sdk.Context, k keeper.Keeper) {
+	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyEndBlocker)
+
+	message := []byte(types.EndBlockSudoMessage)
+
+	p := k.GetParams(ctx)
+
+	errorExecs := make([]string, len(p.ContractAddresses))
+
+	for idx, addr := range p.ContractAddresses {
+		contract, err := sdk.AccAddressFromBech32(addr)
+		if err != nil {
+			errorExecs[idx] = addr
+			continue
+		}
+
+		_, err = k.GetContractKeeper().Sudo(ctx, contract, message)
+		if err != nil {
+			errorExecs[idx] = addr
+			continue
+		}
+
+	}
```

Comment:
Do we want to remove any used gas here from the GasMeter? (Get GasMeter before we iterate, then after each sudo call we get the difference (new-old) and subtract that from new)